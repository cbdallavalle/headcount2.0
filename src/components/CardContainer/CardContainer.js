import React from 'react';
import Card from '../Card/Card';
import PropTypes from 'prop-types';
import astronomy from '../../images/astronomy.svg';
import './CardContainer.css';

const CardContainer = ({schoolData, handleCompareCards}) => {
  const schoolsToRender = Object.keys(schoolData)
    .map((school, index) => <Card
      key={index}
      schoolData={schoolData[school]} 
      handleCompareCards={handleCompareCards}
      size='large'
    />);

  return (
    <div className="CardContainer">
      {schoolsToRender}
      <div className="bottom-images">
        <img src={astronomy} alt='telescope' className='school-icon' />
      </div>
    </div>
  );
};

CardContainer.propTypes = {
  schoolData: PropTypes.oneOfType([
    PropTypes.objectOf(PropTypes.shape({
      location: PropTypes.string.isRequired,
      dataType: PropTypes.string.isRequired,
      data: PropTypes.objectOf(
        PropTypes.number.isRequired)
    })),
    PropTypes.arrayOf(PropTypes.shape({
      location: PropTypes.string.isRequired,
      dataType: PropTypes.string.isRequired,
      data: PropTypes.objectOf(
        PropTypes.number.isRequired)
    }))
  ]),
  handleCompareCards: PropTypes.func.isRequired
};

export default CardContainer;