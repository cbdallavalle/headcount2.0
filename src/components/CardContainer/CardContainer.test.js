/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './CardContainer';

describe('CardContainer', () => {
  let wrapper;

  beforeEach(() => {
    const mockedHandleCompareCards = jest.fn();
    const mockedState = [{
      data: {2004: 0.302, 2005: 0.267, 2006: 0.354, 2007: 0.392, 2008: 0.385,
        2009: 0.39, 2010: 0.436, 2011: 0.489, 2012: 0.479, 2013: 0.488, 2014: 0.49},
      dataType: "Percent",
      location: "ACADEMY 20"}]
    wrapper = shallow(<CardContainer 
                          schoolData={mockedState}
                          handleCompareCards={mockedHandleCompareCards}
                    />)

  })

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  })

  it('should render all the Cards it recieves', () => {
    expect(wrapper.find('Card').length).toEqual(1)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})