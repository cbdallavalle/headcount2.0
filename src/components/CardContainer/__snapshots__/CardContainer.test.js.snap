// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    handleCompareCards={[Function]}
    schoolData={
        Array [
            Object {
              "data": Object {
                "2004": 0.302,
                "2005": 0.267,
                "2006": 0.354,
                "2007": 0.392,
                "2008": 0.385,
                "2009": 0.39,
                "2010": 0.436,
                "2011": 0.489,
                "2012": 0.479,
                "2013": 0.488,
                "2014": 0.49,
              },
              "dataType": "Percent",
              "location": "ACADEMY 20",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <a
            href="#"
>
            <Card
                        handleCompareCards={[Function]}
                        schoolData={
                                    Object {
                                                "data": Object {
                                                  "2004": 0.302,
                                                  "2005": 0.267,
                                                  "2006": 0.354,
                                                  "2007": 0.392,
                                                  "2008": 0.385,
                                                  "2009": 0.39,
                                                  "2010": 0.436,
                                                  "2011": 0.489,
                                                  "2012": 0.479,
                                                  "2013": 0.488,
                                                  "2014": 0.49,
                                                },
                                                "dataType": "Percent",
                                                "location": "ACADEMY 20",
                                              }
                        }
                        size="large"
            />
</a>,
        ],
        <div
          className="bottom-images"
>
          <img
                    alt="telescope"
                    className="school-icon"
                    src="astronomy.svg"
          />
</div>,
      ],
      "className": "CardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <Card
            handleCompareCards={[Function]}
            schoolData={
                        Object {
                                    "data": Object {
                                      "2004": 0.302,
                                      "2005": 0.267,
                                      "2006": 0.354,
                                      "2007": 0.392,
                                      "2008": 0.385,
                                      "2009": 0.39,
                                      "2010": 0.436,
                                      "2011": 0.489,
                                      "2012": 0.479,
                                      "2013": 0.488,
                                      "2014": 0.49,
                                    },
                                    "dataType": "Percent",
                                    "location": "ACADEMY 20",
                                  }
            }
            size="large"
/>,
          "href": "#",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleCompareCards": [Function],
            "schoolData": Object {
              "data": Object {
                "2004": 0.302,
                "2005": 0.267,
                "2006": 0.354,
                "2007": 0.392,
                "2008": 0.385,
                "2009": 0.39,
                "2010": 0.436,
                "2011": 0.489,
                "2012": 0.479,
                "2013": 0.488,
                "2014": 0.49,
              },
              "dataType": "Percent",
              "location": "ACADEMY 20",
            },
            "size": "large",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="telescope"
            className="school-icon"
            src="astronomy.svg"
/>,
          "className": "bottom-images",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "telescope",
            "className": "school-icon",
            "src": "astronomy.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <a
              href="#"
>
              <Card
                            handleCompareCards={[Function]}
                            schoolData={
                                          Object {
                                                        "data": Object {
                                                          "2004": 0.302,
                                                          "2005": 0.267,
                                                          "2006": 0.354,
                                                          "2007": 0.392,
                                                          "2008": 0.385,
                                                          "2009": 0.39,
                                                          "2010": 0.436,
                                                          "2011": 0.489,
                                                          "2012": 0.479,
                                                          "2013": 0.488,
                                                          "2014": 0.49,
                                                        },
                                                        "dataType": "Percent",
                                                        "location": "ACADEMY 20",
                                                      }
                            }
                            size="large"
              />
</a>,
          ],
          <div
            className="bottom-images"
>
            <img
                        alt="telescope"
                        className="school-icon"
                        src="astronomy.svg"
            />
</div>,
        ],
        "className": "CardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <Card
              handleCompareCards={[Function]}
              schoolData={
                            Object {
                                          "data": Object {
                                            "2004": 0.302,
                                            "2005": 0.267,
                                            "2006": 0.354,
                                            "2007": 0.392,
                                            "2008": 0.385,
                                            "2009": 0.39,
                                            "2010": 0.436,
                                            "2011": 0.489,
                                            "2012": 0.479,
                                            "2013": 0.488,
                                            "2014": 0.49,
                                          },
                                          "dataType": "Percent",
                                          "location": "ACADEMY 20",
                                        }
              }
              size="large"
/>,
            "href": "#",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleCompareCards": [Function],
              "schoolData": Object {
                "data": Object {
                  "2004": 0.302,
                  "2005": 0.267,
                  "2006": 0.354,
                  "2007": 0.392,
                  "2008": 0.385,
                  "2009": 0.39,
                  "2010": 0.436,
                  "2011": 0.489,
                  "2012": 0.479,
                  "2013": 0.488,
                  "2014": 0.49,
                },
                "dataType": "Percent",
                "location": "ACADEMY 20",
              },
              "size": "large",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="telescope"
              className="school-icon"
              src="astronomy.svg"
/>,
            "className": "bottom-images",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "telescope",
              "className": "school-icon",
              "src": "astronomy.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
